src/xtopcom/xstate_sync/src/xstate_sync.cpp

```diff
@@ -333,19 +286,19 @@ void xtop_state_sync::fill_tasks(uint32_t n, state_req & req, std::vector<xhash2
         auto const & unit_keys = std::get<2>(res);
         for (size_t i = 0; i < nodes.size(); i++) {
             xdbg("xtop_state_sync::fill_tasks push missing node: %s", nodes[i].as_hex_str().c_str());
-            m_trie_tasks.insert({nodes[i], {}});
+            m_trie_tasks.insert(nodes[i]);
         }
         for (size_t i = 0; i < unit_keys.size(); i++) {
             xdbg("xtop_state_sync::fill_tasks push missing unit: %s", to_hex(unit_keys[i]).c_str());
-            m_unit_tasks.insert({unit_hashes[i], {unit_keys[i], std::set<std::string>()}});
+            m_unit_tasks.insert({unit_hashes[i], unit_keys[i]});
```

Comment:
unit_hashes is always same size as unit_keys?

Reply:
yes, added assert to verify